src/lib/API/api.js

```diff
@@ -207,13 +225,55 @@ class API {
   }
 
   /**
-   * `/verify/facerecognition` get api call
-   * @param {Credentials} creds
+   * `/storage/login/token` get api call
    */
   getLoginToken() {
     return this.client.get('/storage/login/token')
   }
+
+  /**
+   * `/w3Site/api/wl/user` get user from web3 by token
+   * @param {string} token
+   */
+  getUserFromW3ByToken(token: string): Promise<$AxiosXHR<any>> {
+    this.w3Client.defaults.headers.common.Authorization = token
+
+    return this.w3Client.get('/api/wl/user')
+  }
+
+  /**
+   * `/w3Site/api/wl/user` get user from web3 by token
+   * @param {string} token
+   * @param {string} walletAddress
+   */
+  updateW3UserWithWallet(token, walletAddress: string): Promise<$AxiosXHR<any>> {
+    this.w3Client.defaults.headers.common.Authorization = token
+
+    return this.w3Client.put('/api/wl/user/update_profile', {
+      wallet_address: walletAddress,
+    })
+  }
+
+  /**
+   * `/verify/w3/email` verify if user not trying to send some different email than w3 provides
+   * @param {object} data - Object with email and web3 token
+   */
+  checkWeb3Email(data: { email: string, token: string }): Promise<$AxiosXHR<any>> {
+    return this.client.post('/verify/w3/email', data)
+  }
+
+  /**
+   * Get array buffer from image url
+   * @param {string} url - image url
+   */
+  getArrayBufferFromImageUrl(url: string) {
+    return fetch(url, {
```

Comment:
use axios any extra package we use is bad for loading times
```
return axios.get('http://example.com/image.png', { responseType: 'arraybuffer' })
      .then((response) => {
        let image = btoa(
          new Uint8Array(response.data)
            .reduce((data, byte) => data + String.fromCharCode(byte), '')
        );
        return `data:${response.headers['content-type'].toLowerCase()};base64,${image}`;
      });
```

Reply:
done